apm-agent-core/src/main/java/co/elastic/apm/agent/util/BinaryHeaderMap.java

```diff
@@ -71,7 +80,15 @@ public boolean isEmpty() {
         return size() == 0;
     }
 
-    public boolean add(String key, byte[] value) {
+    public boolean add(String key, @Nullable byte[] value) {
+        boolean result;
+        if (value == null) {
+            int size = keys.size();
+            keys.add(key);
+            valueLengths[size] = -1;
```

Comment:
Make sure to `enlargeValueLengths()` if required to avoid `ArrayIndexOutOfBoundsException`.
Better yet, extract this code into a method to be reused with the same done below, something like:
```
int size = keys.size(); 
keys.add(key); 
if (size == valueLengths.length) {
  enlargeValueLengths();
}
valueLengths[size] = length;
```

Reply:
now properly fixed & tested.